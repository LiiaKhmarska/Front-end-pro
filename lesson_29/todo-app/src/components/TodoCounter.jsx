import { useSelector } from 'react-redux';

export function TodoCounter() {

    const counter = useSelector((state) => state.todos.todoCounter)

    return (
        <div className="todo-counter">
            <div>Total: {counter} </div>
        </div>
    )
}