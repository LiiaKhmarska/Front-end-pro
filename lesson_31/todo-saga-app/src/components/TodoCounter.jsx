
export function TodoCounter({counter}) {


    return (
        <div className="todo-counter">
            <div>Total: {counter} </div>
        </div>
    )
}