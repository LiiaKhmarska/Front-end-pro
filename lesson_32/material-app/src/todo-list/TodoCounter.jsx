import { Box, Typography } from '@mui/material';
import { useSelector } from 'react-redux';

export function TodoCounter() {

    const counter = useSelector((state) => state.todos.todoCounter)

    return (
        <Box sx={{textAlign: 'left'}}>
            <Typography sx={{mx:4, fontSize: 24, color: '#1a237e'}}>Total: {counter} </Typography>
        </Box>
    )
}