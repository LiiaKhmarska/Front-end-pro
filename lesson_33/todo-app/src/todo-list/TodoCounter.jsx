import { Box, Typography } from '@mui/material';
import { useSelector } from 'react-redux';
import { selectTodos } from '../store/todoSlice';


export function TodoCounter() {

    const todos = useSelector(selectTodos);
    const counter = todos.length;

    return (
        <Box sx={{textAlign: 'left'}}>
            <Typography sx={{mx:4, fontSize: 24, color: '#1a237e'}} data-testid='todo-counter'>Total: {counter} </Typography>
        </Box>
    )
}