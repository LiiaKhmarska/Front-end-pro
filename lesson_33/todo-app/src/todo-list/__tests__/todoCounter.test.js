import { render, getByTestId } from "@testing-library/react";
import { describe, expect, jest } from "@jest/globals";
import { TodoCounter } from "../TodoCounter";
import { useSelector } from "react-redux";

jest.mock("react-redux");

const mockedInitialState = {
  todos: [{ id: 2147, content: "Make todo-list", checked: true }],
};

describe("Todo Counter render correctly", () => {
  it("todo counter present with default value", () => {
    useSelector.mockReturnValue(mockedInitialState.todos);
    const counter = render(<TodoCounter />);
    expect(counter).toMatchSnapshot();
  });
});
