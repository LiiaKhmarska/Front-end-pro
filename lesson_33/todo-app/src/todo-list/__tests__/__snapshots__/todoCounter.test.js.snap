// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Todo Counter render correctly todo counter present with default value 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-13brihr"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-9fxu5-MuiTypography-root"
          data-testid="todo-counter"
        >
          Total: 
          1
           
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-13brihr"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-9fxu5-MuiTypography-root"
        data-testid="todo-counter"
      >
        Total: 
        1
         
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
